caps/management/commands/import_related_searches.py

```diff
@@ -0,0 +1,91 @@
+"""
+Importer for related search terms
+"""
+
+from pathlib import Path
+
+import requests
+from django.core.management.base import BaseCommand
+from tqdm import tqdm
+
+from caps.models import CachedSearch, KeyPhrase, KeyPhrasePairWise
+from caps.search_funcs import (
+    condense_highlights,
+    fuller_highlighter_config,
+    get_semantic_query,
+)
+
+
+def fetch_data():
+    """
+    Download the data from the cape_ml_search repo
+    """
+
+    keyphrase_url = "https://raw.githubusercontent.com/mysociety/cape_ml_search/main/data/final/ml_keyphrases_with_highlights.csv"
+    pairwise_url = "https://raw.githubusercontent.com/mysociety/cape_ml_search/main/data/final/ml_keyphrases_pairwise.csv"
+
+    keyphrase_file = requests.get(keyphrase_url)
+    pairwise_file = requests.get(pairwise_url)
+
+    Path("data", "ml_keyphrases_with_highlights.csv").write_bytes(
+        keyphrase_file.content
+    )
+    Path("data", "ml_keyphrases_pairwise.csv").write_bytes(pairwise_file.content)
+
+
+def run_searches():
+    """
+    For each primary related search term, run the search in haystack and save the results
+    """
+    new_searches = []
+    allowed_keyphrases = list(KeyPhrase.valid_keyphrases())
+    print(f"Checking {len(allowed_keyphrases)} keyphrases")
+    for keyphrase in tqdm(allowed_keyphrases):
+        tqdm.write(f"Running search for {keyphrase.keyphrase}")
+        sqs, _ = get_semantic_query(keyphrase.keyphrase)
+        results = sqs.highlight(**fuller_highlighter_config)
+        tqdm.write(f"Found {len(results)} matched documents")
+
+        results, _ = condense_highlights(results, [])
+
+        for r in results:
+            doc_id = r.object.id
+
+            # get number of times phrases were found in document
+            if "text" in r.highlighted:
+                highlighted = " ".join(r.highlighted["text"])
+                # count the number of marks
+                count = highlighted.count("<mark>")
+            else:
+                count = 0
+            tqdm.write(f"{doc_id} {count} {keyphrase.keyphrase}")
+            if count > 0:
+                new_searches.append(
+                    CachedSearch(search_term=keyphrase, document_id=doc_id, count=count)
+                )
+
+    CachedSearch.objects.all().delete()
```

Comment:
I was struggling to find a way to query inside a specific document (despite it obviously knowing about individual documents) - but yes, if you figure that out, this could be limited to a subset fairly easily.